django_airavata/apps/groups/views.py

```diff
@@ -2,22 +2,174 @@
 from django.conf import settings
 from django.contrib.auth.decorators import login_required
 from django.shortcuts import render, redirect
+from .forms import CreateForm, AddForm, RemoveForm
+from django.contrib import messages
+from apache.airavata.model.sharing.ttypes import UserGroup
+from apache.airavata.model.sharing.ttypes import GroupCardinality
+import datetime
+import uuid
 
 import logging
 
 logger = logging.getLogger(__name__)
 
+@login_required
+def groups_manage(request):
+
+    gateway_id = settings.GATEWAY_ID
+    username = request.user.username
+
+    try:
+        owner_list = request.sharing_client.getGroups(gateway_id, 0, -1)
+        owner = []
+        for group in owner_list:
+            if group.ownerId == username:
+                owner.append(group)
+        member_list = request.sharing_client.getAllMemberGroupsForUser(gateway_id, username)
+        member = []
+        for group in member_list:
+            if group.ownerId != username:
+                member.append(group)
+        return render(request, 'django_airavata_groups/groups_manage.html', {
+            'owner': owner, 'member': member
+        })
+    except:
+        logger.exception("Failed to load the Manage Groups page")
+        return redirect('/')
 
 @login_required
 def groups_list(request):
 
     gateway_id = settings.GATEWAY_ID
 
     try:
-        groups = request.sharing_client.getGroups(gateway_id, 0, -1)
+        group_list = request.sharing_client.getGroups(gateway_id, 0, -1)
+        for group in group_list:
+            if group.groupCardinality == 0:
+                group_list.remove(group)
         return render(request, 'django_airavata_groups/groups_list.html', {
-            'groups': groups
+            'list': group_list
         })
     except Exception as e:
-        logger.exception("Failed to load groups")
+        logger.exception("Failed to load the list of groups")
         return redirect('/')
+
+@login_required
+def groups_create(request):
+
+    gateway_id = settings.GATEWAY_ID
+    username = request.user.username
+    #If role = user
+    group_type = 1
+    #Else if role = admin
+    #group_type = 2
+    group_cardinality = GroupCardinality.MULTI_USER
+
+    if request.method == 'POST':
+        form = CreateForm(request.POST, request.FILES, initial={'domain_id': gateway_id, 'group_owner': username, 'group_type': group_type, 'group_cardinality': group_cardinality})
+        if form.is_valid():
+            group_id = str(uuid.uuid4().hex) + form.cleaned_data.get('group_name')
```

Comment:
Put the group name first, then the UUID. Also can you replace any spaces in the group_name with hyphens when constructing the group id? You could do something similar like what you did for making the gateway id from the gateway name.